src/zkevm_specs/evm/execution/create.py

```diff
@@ -0,0 +1,161 @@
+    # the last one (GAS_COST_CODE_DEPOSIT * len(byte_code)) would be calculated in `return_revert`
+    gas_left = instruction.curr.gas_left
+    gas_cost = GAS_COST_CREATE + memory_expansion_gas_cost
+    gas_available = gas_left - gas_cost
+
+    # Apply EIP 150
+    one_64th_gas, _ = instruction.constant_divmod(gas_available, FQ(64), N_BYTES_GAS)
+    all_but_one_64th_gas = gas_available - one_64th_gas
+    is_u64_gas = instruction.is_zero(
+        instruction.sum(instruction.rlc_encode(gas_left, N_BYTES_GAS).le_bytes[N_BYTES_GAS:])
+    )
+    callee_gas_left = instruction.select(
+        is_u64_gas,
+        instruction.min(all_but_one_64th_gas, gas_left, N_BYTES_GAS),
+        all_but_one_64th_gas,
+    )
+
+    # copy init_code from memory to bytecode
+    copy_rwc_inc, _ = instruction.copy_lookup(
+        instruction.curr.call_id,  # src_id
+        CopyDataTypeTag.Memory,  # src_type
+        code_hash.expr(),  # dst_id
+        CopyDataTypeTag.Bytecode,  # dst_type
+        instruction.rlc_to_fq(offset, N_BYTES_U64),  # src_addr
+        instruction.rlc_to_fq(offset, N_BYTES_U64)
+        + instruction.rlc_to_fq(size, N_BYTES_U64),  # src_addr_boundary
+        FQ(0),  # dst_addr
+        size,  # length
+        instruction.curr.rw_counter + instruction.rw_counter_offset,
+    )
+    instruction.rw_counter_offset += int(copy_rwc_inc)
+
+    # CREATE:  3 pops and 1 push, stack delta = 2
+    # CREATE2: 4 pops and 1 push, stack delta = 3
+    stack_pointer_delta = 2 + is_create2
+    # Save caller's call state
+    for field_tag, expected_value in [
+        (CallContextFieldTag.ProgramCounter, instruction.curr.program_counter + 1),
+        (
+            CallContextFieldTag.StackPointer,
+            instruction.curr.stack_pointer + stack_pointer_delta,
+        ),
+        (CallContextFieldTag.GasLeft, gas_left - gas_cost - callee_gas_left),
+        (CallContextFieldTag.MemorySize, next_memory_size),
+        (
+            CallContextFieldTag.ReversibleWriteCounter,
+            instruction.curr.reversible_write_counter + 1,
+        ),
+    ]:
+        instruction.constrain_equal(
+            instruction.call_context_lookup(field_tag, RW.Write),
+            expected_value,
+        )
+
+    # Setup next call's context.
+    for field_tag, expected_value in [
+        (CallContextFieldTag.CallerId, instruction.curr.call_id),
+        (CallContextFieldTag.TxId, tx_id.expr()),
+        (CallContextFieldTag.Depth, depth.expr() + 1),
+        (CallContextFieldTag.CallerAddress, caller_address.expr()),
+        (CallContextFieldTag.CalleeAddress, contract_address.expr()),
+        (CallContextFieldTag.IsSuccess, FQ(True)),
+        (CallContextFieldTag.IsStatic, FQ(False)),
+        (CallContextFieldTag.IsRoot, FQ(False)),
+        (CallContextFieldTag.IsCreate, FQ(False)),
+        # FIXME change to assign either 0 or EMPTY_CODE_HASH? or only EMPTY_CODE_HASH
+        (CallContextFieldTag.CodeHash, RLC(EMPTY_CODE_HASH)),
```

Comment:
Not sure which we should set to the next context. Set what it is in caller's context or a fixed value, `EMPTY_CODE_HASH` is enough? have any ideas, @han0110 ?

Reply:
Next's call context's `code_hash` will just be the initcode's hash, in case the initcode could still switch into deeper context, we need to setup beforehand.

Reply:
fixed in [e8d9f05](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/e8d9f05bc2a014be74442e84b907966de3af68e1)